### PR TITLE
Add AzureDisk CSI driver installation

### PR DESCRIPTION
**Pull request by jsafrane**
As tech preview.

cc @openshift/storage

### CODE REVIEW

**Comment by jsafrane** [comment]
/hold
for manual testing

**Comment by jsafrane** [comment]
/hold cancel
Tested manually, installs the driver when TechPreviewNoUpgrade featureSet is set,
I'm curious about azure CI job - it does not install the driver, it just tells us we did not break cluster installation in this PR.

**Comment by jsafrane** [comment]
/hold
for the CI to finish

File: assets/csidriveroperators/azure-disk/06_clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  - list
+  - watch
+  - update
+  - delete
+- apiGroups:
+  - cloudcredential.openshift.io
+  resources:
+  - credentialsrequests
+  verbs:
+  - '*'
```
Comment (bertinatto) [inline assets/csidriveroperators/azure-disk/06_clusterrole.yaml:253]: hm, I think we'll have to remove this from all our CSI operators...

**Comment by jsafrane** [comment]
/hold cancel
Azure installs (but it's quite flaky, does not look like storage issue)

File: assets/csidriveroperators/azure-disk/08_deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  namespace: openshift-cluster-csi-drivers
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: azure-disk-csi-driver-operator
+  strategy: {}
+  template:
+    metadata:
+      labels:
```
Comment (cynepco3hahue) [inline assets/csidriveroperators/azure-disk/08_deployment.yaml:14]: Can you please add the annotation:
```
annotations:
        workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
```
it is critical for our work in support of the workload partitioning feature
(openshift/enhancements#703)

**Pull request merged**